### PR TITLE
fix Value of type null is not callable error if $next is null and not callable

### PR DESCRIPTION
**Pull request by DyarWeb**
When the {next} variable is not set, we receive an error: Value of type null is not callable
And this causes the robot not to work properly

I just added a condition to check if the {next} variable is callable or not, if not, null will be returned

### CODE REVIEW

**Comment by DyarWeb** [comment]
Change the PHP version Because the new version of the project uses the Typed property and because this feature has been added to later versions 7.4, we will get an error in version 7.2 and it will not work.

File: src/Handlers/UpdateHandler.php [hunk 1]
```diff
@@ -86,6 +86,6 @@ public function __invoke($next)
             $result = $this->handle();
         }
 
-        return $result ?? $next();
+        return $result ?? (is_callable($next) ? $next() : null);
```
Comment (punyflash) [inline src/Handlers/UpdateHandler.php:89]: There is no such case actually when the `$next = null`. The runner should always yield and return callable, so I wonder how you end up having this issue: https://github.com/westacks/telebot/blob/363fc68d4fc6c8c60be98e6ce61ef99448582037/src/Kernel.php#L69-L80

Please provide how you end up with it, and I'll accept it gladly if this condition is necessary.

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
         }
     ],
     "require": {
-        "php": "^7.2 || ^8.0",
+        "php": "^7.4 || ^8.0",
```
Comment (punyflash) [inline composer.json:15]: I don't mind dropping 7.2-7.3 as they out of support already, so ok.

**Comment by punyflash** [comment]
So I tested it a bit detailed and the fix is not that simple. The error is located in runner where `$next` was basically called before passing the callable to the handler. The correct runner should be

```php
public function run(TeleBot $bot, Update $update)
    {
        $runner = $this->runner($bot);

        if (!$runner->valid()) {
            return null;
        }

        $start = $runner->current();
        $pipeline = function () use ($runner, $update, &$pipeline) {
            $runner->next();

            return $runner->valid() ?
                $runner->current()($update, $pipeline) :
                $runner->getReturn();
        };

        return $start($update, $pipeline);
    }
```

**Pull request closed without merging**